esphome/components/http_request/__init__.py

```diff
@@ -189,4 +197,9 @@ def http_request_action_to_code(config, action_id, template_arg, args):
         )
         cg.add(var.add_header(key, template_))
 
+    for conf in config.get(CONF_ON_RESPONSE, []):
+        trigger = cg.new_Pvariable(conf[CONF_TRIGGER_ID])
+        cg.add(var.register_response_trigger(trigger))
+        yield automation.build_automation(trigger, [(int, "x")], conf)
```

Comment:
```suggestion
        yield automation.build_automation(trigger, [(int, "status_code")], conf)
```

I assume the plan is to attach the response payload as another variable to this `on_response` automation, right? If so naming the `status_code` using a long name would be better.